From 57452b0e88731d3afe22fe3170f5f64d22ccf3d5 Mon Sep 17 00:00:00 2001
From: Legrand Benjamin <android@legrand.ws>
Date: Fri, 29 Mar 2013 17:25:25 +0100
Subject: [PATCH] [Mediatek build hack]

Add some parameters to the kernel make command so that we can use the Mediatek specific kernel build system

Change-Id: I43ca39617585683efdf0934d806312be7675fa2c
---
 core/tasks/kernel.mk | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/core/tasks/kernel.mk b/core/tasks/kernel.mk
index 7b68726..853fdde 100644
--- a/core/tasks/kernel.mk
+++ b/core/tasks/kernel.mk
@@ -151,16 +151,24 @@ $(KERNEL_OUT):
 	mkdir -p $(KERNEL_OUT)
 	mkdir -p $(KERNEL_MODULES_OUT)
 
+ifeq $($(KERNEL_MAKE_PARAMETERS),)
+    MAKE_ADDITIONAL_PARAMETERS := 
+else
+    MAKE_ADDITIONAL_PARAMETERS := $(KERNEL_MAKE_PARAMETERS)
+endif
+
+
 $(KERNEL_CONFIG): $(KERNEL_OUT)
 	$(MAKE) -C $(KERNEL_SRC) O=$(KERNEL_OUT) ARCH=$(TARGET_ARCH) $(ARM_CROSS_COMPILE) $(KERNEL_DEFCONFIG)
 
 $(KERNEL_OUT)/piggy : $(TARGET_PREBUILT_INT_KERNEL)
 	$(hide) gunzip -c $(KERNEL_OUT)/arch/$(TARGET_ARCH)/boot/compressed/piggy.gzip > $(KERNEL_OUT)/piggy
 
+
 TARGET_KERNEL_BINARIES: $(KERNEL_OUT) $(KERNEL_CONFIG) $(KERNEL_HEADERS_INSTALL)
-	$(MAKE) -C $(KERNEL_SRC) O=$(KERNEL_OUT) ARCH=$(TARGET_ARCH) $(ARM_CROSS_COMPILE) $(TARGET_PREBUILT_INT_KERNEL_TYPE)
-	$(MAKE) -C $(KERNEL_SRC) O=$(KERNEL_OUT) ARCH=$(TARGET_ARCH) $(ARM_CROSS_COMPILE) modules
-	$(MAKE) -C $(KERNEL_SRC) O=$(KERNEL_OUT) INSTALL_MOD_PATH=../../$(KERNEL_MODULES_INSTALL) ARCH=$(TARGET_ARCH) $(ARM_CROSS_COMPILE) modules_install
+	$(MAKE) $(MAKE_ADDITIONAL_PARAMETERS) -C $(KERNEL_SRC) O=$(KERNEL_OUT) ARCH=$(TARGET_ARCH) $(ARM_CROSS_COMPILE) $(TARGET_PREBUILT_INT_KERNEL_TYPE)
+	$(MAKE) $(MAKE_ADDITIONAL_PARAMETERS) -C $(KERNEL_SRC) O=$(KERNEL_OUT) ARCH=$(TARGET_ARCH) $(ARM_CROSS_COMPILE) modules
+	$(MAKE) $(MAKE_ADDITIONAL_PARAMETERS) -C $(KERNEL_SRC) O=$(KERNEL_OUT) INSTALL_MOD_PATH=../../$(KERNEL_MODULES_INSTALL) ARCH=$(TARGET_ARCH) $(ARM_CROSS_COMPILE) modules_install
 	$(mv-modules)
 	$(clean-module-folder)
 
-- 
1.8.1.2

